### DISCUSSION TITLE
more "saveable" settings

### DISCUSSION THREAD

**Discussion by Frank-D-HB** [zwave-js/zwave-js-ui#2214] [Ideas]
Proposal:
1. moving the "Dark theme"-switch to the "Gerneral Settings"-tab, incl. saving the selection, if the save-button there is pushed.
2. in "Network Graph"-tab, add a save-Button to "options-section" for the actual selection.

**Comment by robertsLando** [comment]
@Frank-D-HB the dark theme setting is stored on local storage like all table columns and filters prefrences

About the network graph I could store those settings on localstorage too

**Reply by Frank-D-HB** [reply]
to store the graph settings at local store would be helpful, while "surfing" between control-panel and network-panel.

But after leaving the browser session, the the lokalstorage seems to be erased, so the idea (if its possible without much effort), to store these ui-settings inside zwavejs2mqtt. I would get the same "user experience", if i use different devices to manage zwavejs2mqqt.

**Reply by robertsLando** [reply]
Makes sense. BTW that requires much more work. There is already an open feature request for that: 

https://github.com/zwave-js/zwavejs2mqtt/issues/2058

**Comment by Frank-D-HB** [comment]
I have seen that you store "data" in files like "settings.json" and so on...
Just a new file "UI-Settings.json"(?), where these settings were saved.
There could be saved the general settings like "theme", which "work" at all browsers/"ui-devices", 
and later on it is may be extendable to specific settings-area for firefox, chrome or other "clients".

Another possibility i have seen at other browser bases ui, is to "store" settings together with the url (saving the bookmark).
Example what i mean:  https://my_local_zwavejs2mqtt.com/theme?q=dark&network_grouped?q=by_location

Hopefully i did not blame me, by such proposals, cause i am not a computer expert.
